### ISSUE TITLE
beforeCrop property problem

### ISSUE THREAD

**Issue by Valtas** [nanxiaobei/antd-img-crop#274]
Codesandbox https://codesandbox.io/p/sandbox/antd-img-crop-beforecrop-problem-wftx6c

Open console
Upload a file larger than 2 MB
Called beforeUpload

It is expected that beforeUpload will not be called. This behavior was present in earlier versions of antd-img-crop

**Comment by nanxiaobei** [comment]
yes, the logic has changed,
`beforeCrop` is only used to determine whether to open the modal, 
`beforeUpload` is used to determine whether to upload.

**Comment by Valtas** [comment]
because of this I get a situation where I can’t prohibit files at the img-crop stage

beforeCrop - I check that the file is acceptable for img-crop. The file is invalid but calls beforeUpload, although upload is prohibited...

in beforeUpload I also do a protection check, and I get a double check and a double error message..

**Comment by nanxiaobei** [comment]
yes, this may require check twice.
but there are still some usage scenarios, 
they hope to crop some images, and upload other images directly.
it's difficult to balance.

for your scene, I think this can be done:

```jsx
const canUpload = useRef(false)
const beforeCrop = (file) => {
  // do check
}

beforeCrop={(file) => {
  canUpload.current = beforeCrop(file)
  return canUpload.current
}}
beforeUpload={() => canUpload.current}
```

**Comment by Valtas** [comment]
Yes, there are options to handle the case. It used to be really great)

Perhaps all situations would be resolved if beforeCrop returned an enum value that either completely stopped or skipped the crop stage

enum.SKIP
enum.STOP ...

**Comment by nanxiaobei** [comment]
thx, will consider that.
